Server/tests/controllers/monitorController.test.js

```diff
@@ -473,6 +475,58 @@ describe("Monitor Controller - createMonitor", () => {
   });
 });
 
+describe('checkEndpointResolution', () => {
+  let dnsResolveStub;
+  beforeEach(() => {
+    req = { query: { monitorURL: 'example.com' } };
+    res = { status: sinon.stub().returnsThis(), json: sinon.stub() };
+    next = sinon.stub();
+    dnsResolveStub = sinon.stub(dns, 'resolve');
+  });
+  afterEach(() => {
+    dnsResolveStub.restore();
+  });
+  it('should resolve the URL successfully', async () => {
+    dnsResolveStub.callsFake((monitorURL, callback) => callback(null));
+    await checkEndpointResolution(req, res, next);
+    expect(res.status.calledWith(200)).to.be.true;
+    expect(res.json.calledWith({
+      success: true,
+      msg: 'URL resolved successfully',
+    })).to.be.true;
+    expect(next.called).to.be.false;
+  });
+  it("should return a 400 error message if DNS resolution fails", async () => {
+    const dnsError = new Error("DNS resolution failed");
+    dnsResolveStub.callsFake((monitorURL, callback) => callback(dnsError));
+    await checkEndpointResolution(req, res, next);
+    expect(res.status.calledOnceWith(400)).to.be.true;
+    expect(res.json.calledOnceWith({
+      success: false,
+      msg: "DNS resolution failed",
+    })).to.be.true;
+    expect(next.notCalled).to.be.true;
+  });
+  it("should remove the trailing slash and resolve the URL successfully", async () => {
+    req.query.monitorURL = 'http://example.com/'; // URL with a trailing slash
+    dnsResolveStub.callsFake((monitorURL, callback) => callback(null));
+    await checkEndpointResolution(req, res, next);
+    expect(res.status.calledWith(200)).to.be.true;
+    expect(res.json.calledWith({
+      success: true,
+      msg: 'URL resolved successfully',
+    })).to.be.true;
+    expect(next.called).to.be.false;
+  });
+  it("should call next with an error if an exception is thrown", async () => {
+    req.query = {};  // Missing monitorURL will cause an error
+    await checkEndpointResolution(req, res, next);
+    expect(next.calledOnce).to.be.true;
+    const error = next.getCall(0).args[0];
+    expect(error).to.be.an.instanceOf(Error);
```

Comment:
It would be best to check for the specific error that you are getting rather than just to see if an error is thrown.

We should be very specific, if you had a function that did:

```
dangerousOperationOne()
dangerousOperationTwo()
```

and your test just checks to see if an error is thrown you wouldn't know if it was `dangerousOpeartionOne()` or `dangerousOperationTwo()` that failed